src/main/webapp/portal/search/controllers.js

```diff
@@ -33,18 +31,23 @@ define(['angular'], function(angular) {
       };
 
       $scope.submit = function(){
-          if($scope.initialFilter != "") {
-            if(APP_FLAGS.isWeb) {
-                $location.path("apps/search/"+ $scope.initialFilter);
-                $scope.initialFilter = "";
-                $scope.showSearch = false;
-                $scope.showSearchFocus = false;
-            } else {
-                //frame app redirect
-                window.location = SEARCH.searchURL + $scope.initialFilter;
-            }
+        if($scope.initialFilter != "") {
+          if(APP_FLAGS.isWeb) {
+              $location.path("apps/search/"+ $scope.initialFilter);
+              $scope.showSearch = false;
+              $scope.showSearchFocus = false;
+          } else {
+              //frame app redirect
+              window.location = SEARCH.searchURL + $scope.initialFilter;
           }
-        };
+        }
+      };
+        
+      $scope.$on('$locationChangeStart', function(event) {
+        if ($location.url().indexOf('search') == -1) {
+          $scope.initialFilter = '';
```

Comment:
Seems dangerous if we ever change urls, but I guess we're in a lot of trouble when we start changing any of our urls around.